mozilla/pdf.js
# Improve support for viewing PDFs from file://-URLs in the Chrome extension

Up to now, if a user did not check the "Allow access to file URLs" checkbox at the extension settings page, then they could not view local PDFs in Chromium.

With this patch, such navigations are intercepted, and the extension will ask the user to enable the setting when it detects that it has no access to file:-URLs.

When the user selects "Allow access to file URLs", the extension reloads and Chrome closes all existing extension tabs (https://crbug.com/511670). Since this is disruptive, I've added a work-around (save tab state before page unload, restore tab state when extension loads).

To test:
1. node make chromium
2. chromium --load-extension=build/chromium
3. Visit chrome://extensions and uncheck the 'Allow access to file URLs" at the PDF.js extension (this was automatically checked because you've loaded the extension in unpacked mode).
4. Open a local PDF in the browser.
5. Verify that the PDF.js extension prompts for file access permissions.
6. Follow the instructions:
   - Click on the chrome://extensions link.
   - Click on the "Allow access to file URLs" checkbox at the extensions page.
7. Navigate back
8. Verify that the local PDF is displayed by the extension.<br><br>
9. Go to chrome://extensions in a new tab (so the PDF viewer is still open).
10. Click on "Allow access to file URLs" (to untick the checkbox).
11. Observe that Chrome closes the PDF viewer tab, but verify that the PDF Viewer extension re-opens the tab (and prompts to re-enable file access).

Screenshot:

![Chrome extension asking for access to local files](https://cloud.githubusercontent.com/assets/1365071/8762691/3141a5ac-2d81-11e5-879c-101630f00606.png)

The displayed string adapts itself to the browser. Start Chrome/Chromium with the `LANG=nl_NL.UTF8` environment variable, and you will see a Dutch browser interface and the Dutch translation of "Allow access to file URLs":

![Chrome extension asking for access to local files (LANG=nl_NL.UTF8)](https://cloud.githubusercontent.com/assets/1365071/8762712/7acac154-2d81-11e5-819c-23ba05628e69.png)
